app/models/notification.rb

```diff
@@ -0,0 +1,48 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+
+  belongs_to :user
+  belongs_to :course
+  def self.center_popup(target_course_id, target_user_id, options = {})
+    notification = Notification.new
+    notification.course_id = target_course_id
+    notification.user_id = target_user_id
+    set_options(notification, options)
+    notification.n_type = 'center_popup'
+    notification.save
+  end
+
+  def self.right_side_popup(target_course_id, target_user_id, options = {})
+    notification = Notification.new
+    notification.course_id = target_course_id
+    notification.user_id = target_user_id
+    set_options(notification, options)
+    notification.n_type = 'right_side_popup'
+    notification.save
+  end
+
+  def self.activity_only(target_course_id, target_user_id, options = {})
+    notification = Notification.new
+    notification.course_id = target_course_id
+    notification.user_id = target_user_id
+    notification.action = options[:action]
+    notification.activity_content = options[:activity_content]
+    notification.link = options[:link]
+    notification.save
+  end
+
+  def self.set_options(notification, options)
+    notification.action = options[:action]
+    notification.activity_content = options[:activity_content]
+    notification.n_title = options[:n_title]
+    notification.n_content = options[:n_content]
+    notification.image = options[:image]
+    notification.link = options[:link]
+    notification.share = options[:share]
+  end
+
+  def self.recent_activity(course_id, number)
```

Comment:
I prefer a single unified API. You can create it using `Notification.create(..., types: [:recent_activity, :popup, ...])` The API above restricts you to just one kind.